staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -101,7 +101,8 @@ func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *v1beta
 		return &webhook.ErrCallingWebhook{WebhookName: h.Name, Reason: err}
 	}
 	response := &admissionv1beta1.AdmissionReview{}
-	if err := client.Post().Context(ctx).Body(&request).Do().Into(response); err != nil {
+	if err := client.Post().Context(ctx).Timeout(time.Duration(*h.TimeoutSeconds) * time.Second).
```

Comment:
only call Timeout if h.TimeoutSeconds is non-nil?

Reply:
talked with @mbohlool about this

- in apiserver every `h.TimeoutSeconds` must be non-nil since we do defaulting, so we could assume non-nil and panic/fail hard when we see nil value
- we do have [tests](https://github.com/kubernetes/kubernetes/blob/493b26105789bf88575b78cce9e8474e20ae3f48/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/plugin_test.go#L52) that don't go through apiserver defaulting and call Dispatch directly with nil-value timeout, so checking non-nil is more backwards compatible for us (or we can always fix those tests)

I don't feel strongly either way. Updated